libr/asm/arch/8051/8051_ass.c

```diff
@@ -1040,6 +1073,31 @@ static bool mnem_mov(char const*const*arg, ut16 pc, ut8**out) {
 		*out += 3;
 		return true;
 	}
+
+	if (!r_str_casecmp (arg[0], "@r0")) {
+		ut16 imm;
+		if (!resolve_immediate (arg[1] + 1, &imm)) {
+			return false;
+		}
+		(*out)[0] = 0x76;
+		(*out)[1] = imm >> 8;
+		(*out)[2] = imm;
+		*out += 3;
+		return true;
+	}
+
+	if (!r_str_casecmp (arg[0], "@r1")) {
+		ut16 imm;
+		if (!resolve_immediate (arg[1] + 1, &imm)) {
+			return false;
+		}
+		(*out)[0] = 0x77;
+		(*out)[1] = imm >> 8;
+		(*out)[2] = imm;
+		*out += 3;
```

Comment:
out can overflow, you should pass the buffer size before passing it to this function